### PR TITLE
Fix #76 resources names

### LINKED ISSUE #76: Cannot convert undefined or null to object.
Attempting to run API Platform client generation on a very basic API Platform with no particularly complex entities set up, running into this error when attempting to parse with OpenAPI version 3:

```
TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at /home/node/.npm/_npx/233/lib/node_modules/@api-platform/client-generator/node_modules/@api-platform/api-doc-parser/lib/openapi3/handleJson.js:19:34
    at Array.map (<anonymous>)
    at Object.default_1 [as default] (/home/node/.npm/_npx/233/lib/node_modules/@api-platform/client-generator/node_modules/@api-platform/api-doc-parser/lib/openapi3/handleJson.js:16:27)
    at /home/node/.npm/_npx/233/lib/node_modules/@api-platform/client-generator/node_modules/@api-platform/api-doc-parser/lib/openapi3/parseOpenApi3Documentation.js:12:45
    at processTicksAndRejections (internal/process/task_queues.js:94:5)
```

the JSON I'm handing it, which is autogenerated by API Platform, is valid and the OpenAPI spec says it's completely fine, except it's missing "description" fields on the properties (though these aren't necessary by my understanding).

Given the relative opacity of this error it's not clear where it's failing to parse my JSON. I've spent several hours spinning my wheels on this with no progress, and I'm starting to feel quite bad about it, so any help would be much appreciated.

### PR DESCRIPTION
**Pull request by soyuka**
see https://github.com/api-platform/api-doc-parser/pull/86 and #76

### CODE REVIEW

File: src/utils/getResources.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export const getResources = (paths: any) => {
+  const resources: string[] = [];
+
+  Object.keys(paths).forEach((item, index, array) => {
+    if (item.includes("/{id}")) return;
+    if (resources.find(path => path.startsWith("/" + item.split("/")[1])))
```
Comment (jamesisaac) [inline src/utils/getResources.ts:6]: This will return different results depending on what order `/test` and `/test/bar` are encountered.

**Comment by jamesisaac** [comment]
@soyuka Whoops sorry got in a bit too late with my comment

**Comment by soyuka** [comment]
> @soyuka Whoops sorry got in a bit too late with my comment

Nooo it's my bad I shouldn't have rushed! You're right about the `paths[path][method][tags][0]` thing but we then need to find a way for https://github.com/api-platform/api-doc-parser/blob/master/src/openapi3/handleJson.ts#L23-L25

IMO this library is good to handle resources with basic CRUD. I'm afraid to change this as you said

> I guess that would be an API break for any tools you have depending on the output from this though? So not sure how you'd want to handle that sort of change.

Could you explain what is your usage of this library?

**Comment by jamesisaac** [comment]
> Could you explain what is your usage of this library?

No worries, I was just trying this library out for some custom client-side typedef generation.  But it looks like this one is more built around the design of having 1 type per Resource which lists all the possible readable/writeable fields.  Whereas I wanted one exact typedef per operation.  So in the end I just wrote my own parser.

If this lib (and the ecosystem of others that rely on it, like the API Platform admin generator) are more geared towards basic CRUD, then sounds like it probably makes more sense to leave it as it is.

On another note though I did encounter a couple of bugs in the OpenAPI doc generation of API Platform, which any parser would benefit from having fixed.  I'll open issues for those in the main repo once I've had time to prepare minimal repros.

**Pull request merged**